src/app/user.spec.ts

```diff
@@ -0,0 +1,41 @@
+import {User} from './user';
+import {UserData} from './user-data';
+
+describe('User', () => {
+  let user: User;
+  beforeEach(() => {
+    user = new User();
+  });
+  it('should create an instance', () => {
+    expect(user).toBeTruthy();
+  });
+  it('should get Id', () => {
+    expect(user.getId()).toEqual(-1);
+  });
+  it('should get email', () => {
+    expect(user.getEmail()).toEqual('default');
+  });
+  it('should get password', () => {
+    expect(user.getPassword()).toEqual('default');
+  });
+  it('should get userData', () => {
+    expect(user.getUserData()).toBeTruthy();
+  });
+  it('should set Id', () => {
+    user.setId(0);
+    expect(user.getId()).toEqual(0);
+  });
+  it('should set email', () => {
+    user.setEmail('notDefault');
+    expect(user.getEmail()).toEqual('notDefault');
+  });
+  it('should set password', () => {
+    user.setPassword('notDefault');
+    expect(user.getPassword()).toEqual('notDefault');
+  });
+  it('should set userData', () => {
+    const userDataNotDefault = new UserData('Fabio');
+    user.setUserData(userDataNotDefault);
+    expect(user.getUserData()).toEqual(userDataNotDefault);
+  });
```

Comment:
questi test tiran su la coverage ma sono abbastanza ridondanti; nel senso che controllano feature basilari del linguaggio più che della classe in se.

piuttosto farei un test unico in cui vengono cambiate tutte le proprietà rilevanti e si fa l'expect sull'oggetto, tipo:

```ts
const expectedProps = {
 firstName: "expectedFirstName",
....
}
 // questa sintassi spacchetta expectedProps e chiama correttamente il costruttore
const expectedUser = new User( ...expectedProps)

user.setFirstName(expectedProps.firstName);
// ...
expect(user).toEqual(expectedUser); // bisogna controllare se toEqual controlla property-by-property
```

Reply:
> expect(user).toEqual(expectedUser); // bisogna controllare se toEqual controlla property-by-property

sono d'accordo

Reply:
si avevamo testato anche noi che  che toEqual è molto più intelligente di quanto pensassimo


w